packages/cli/src/build/assets.rs

```diff
@@ -0,0 +1,320 @@
+use std::{
+    io::{Cursor, Read, Seek, Write},
+    path::{Path, PathBuf},
+};
+
+use crate::Result;
+use anyhow::Context;
+use const_serialize::{ConstVec, SerializeConst};
+use dioxus_cli_opt::AssetManifest;
+use manganis::BundledAsset;
+use object::{File, Object, ObjectSection, ObjectSymbol, ReadCache, ReadRef, Section, Symbol};
+use pdb::FallibleIterator;
+use rayon::iter::{IntoParallelRefMutIterator, ParallelIterator};
+use wasmparser::BinaryReader;
+
+/// Extract all manganis symbols and their sections from the given object file.
+fn manganis_symbols<'a, 'b, R: ReadRef<'a>>(
+    file: &'b File<'a, R>,
+) -> impl Iterator<Item = (Symbol<'a, 'b, R>, Section<'a, 'b, R>)> + 'b {
+    file.symbols()
+        .filter(|symbol| {
+            if let Ok(name) = symbol.name() {
+                name.contains("__MANGANIS__")
+            } else {
+                false
+            }
+        })
+        .filter_map(move |symbol| {
+            let section_index = symbol.section_index()?;
+            let section = file.section_by_index(section_index).ok()?;
+            Some((symbol, section))
+        })
+}
+
+/// Find the offsets of any manganis symbols in the given file.
+fn find_symbol_offsets<'a, R: ReadRef<'a>>(
+    path: &Path,
+    file_contents: &[u8],
+    file: &File<'a, R>,
+) -> Result<Vec<u64>> {
+    let pdb_file = find_pdb_file(path);
+
+    if file.format() == object::BinaryFormat::Wasm {
+        find_wasm_symbol_offsets(file_contents, file)
+    } else if let Some(pdb_file) = pdb_file {
+        find_pdb_symbol_offsets(&pdb_file)
+    } else {
+        find_native_symbol_offsets(file)
+    }
+}
```

Comment:
maybe match on the binary format / os like we do in patch.rs?

Reply:
I don't think we have the OS info here if you run the asset command directly, and you might be cross-compiling, so we shouldn't rely on the host os